### PR TITLE
add plugin usage

### PR DESCRIPTION
**Pull request by shoota**
# TL;DR

Connect to #106

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ npm install @userdive/agent --save
 
 ## Usage
 
-Standard Usage
+#### Standard Usage
```
Comment (9renpoto) [inline README.md]: Basic

File: README.md [hunk 2]
```diff
@@ -67,6 +67,32 @@ agent.set('set', {
...
 
+#### Register Plugin ( in plugin code )
+
+```js
+export MyPlugin = function() {
+	this.say = function ( msg ) {
+		console.log( 'say: ' + msg )
+	}
+}
+agent.provide('myplugin', MyPlugin)
```
Comment (9renpoto) [inline README.md]: MyPlugin defined `class` based syntax

File: README.md [hunk 3]
```diff
@@ -67,6 +67,32 @@ agent.set('set', {
 agent.set('set', 'dimension15', 'My Custom Dimension')
 ```
 
+#### Register Plugin ( in plugin code )
+
+```js
+export MyPlugin = function() {
+	this.say = function ( msg ) {
```
Comment (9renpoto) [inline README.md]: goodbye tab indent

File: README.md [hunk 4]
```diff
@@ -67,6 +67,32 @@ agent.set('set', {
...
+#### Use Plugin
+
+plugin function can call `'pluginName:functionName'`
+
+```js
+import Agent from '@userdive/agent'
+const agent = new Agent()
+agent.create('projectId', 'auto')
+agent.require('myplugin')  // provided name
+agent.run('myplugin:say', 'hello')
```
Comment (9renpoto) [inline README.md]: what is run namespace

File: README.md [hunk 5]
```diff
@@ -67,6 +67,36 @@ agent.set('set', {
...
 
+#### Register Plugin ( in plugin code )
+
+```js
+class MyPlugin {
+  constructor (agent, opts) {
+    this.opts = opts || {msg: ''}
+  }
+
+  say () {
```
Comment (9renpoto) [inline README.md]: static say ?

Comment (shoota) [reply README.md]: oh...

Comment (shoota) [reply README.md]: it is class member function. not static.

**Pull request merged**